jobs/build/rhcos_sync/Jenkinsfile

```diff
@@ -119,19 +120,34 @@ node {
         }
 
         tag = params.FROM_RELEASE_TAG
-        if (params.OVERRIDE_NAME != "") {
-            name = params.OVERRIDE_NAME
-        } else {
-            name = tag
+        image = "quay.io/openshift-release-dev/ocp-release:${tag}"
+        if (tag.contains("/")) {
+            // assume instead of a tag it's a pullspec e.g. to registry.ci
+            if (!tag.contains(":")) error("FROM_RELEASE_TAG pullspec must include a :tag")
+            image = tag
+            tag = tag.split(":")[-1]
         }
 
+        if (params.OVERRIDE_NAME != "")
+            name = params.OVERRIDE_NAME
+        else
+            name = commonlib.shell(
+                returnStdout: true,
+                script: "oc adm release info -o template --template '{{ .metadata.version }}' ${image}"
+            )
+
         (major, minor) = commonlib.extractMajorMinorVersionNumbers(tag)
         ocpVersion = "$major.$minor"
 
         (arch, priv) = releaselib.getReleaseTagArchPriv(tag)
         suffix = releaselib.getArchPrivSuffix(arch, priv)
 
-        cmd = "oc image info -o json \$(oc adm release info --image-for machine-os-content registry.ci.openshift.org/ocp$suffix/release$suffix:$tag) | jq -r .config.config.Labels.version"
+        cmd = """
+            tmp=\$(mktemp -d /tmp/tmp.XXXXXX)
+            oc image extract --path /manifests/:\$tmp \$(oc adm release info --image-for installer ${image})
+            cat \$tmp/coreos-bootimages.yaml | yq -r .data.stream | jq -r .architectures.${arch}.artifacts.qemu.release
```

Comment:
Yes, the installer wraps the JSON... in a YAML definition of an API object. I seriously considered just using the source label and looking it up from github. There are two issues with that though:
1. Requires an access token for github and other complications
2. Requires consulting github when the data is right there in the payload... I'd rather use the source of truth.